core-concepts/datanft-and-datatoken.md

```diff
@@ -43,6 +43,16 @@ Here's an example.
 * In step 1, Alice **publishes** her dataset with Ocean: this means deploying an ERC721 data NFT contract (claiming copyright/base IP), then an ERC20 datatoken contract (license against base IP).
 * In step 2, she **mints** some ERC20 datatokens and **transfers** 1.0 of them to Bob's wallet; now he has a license to be able to download that dataset.
 
+### TemplateIds
+
+Each data NFT or a datatoken is cloned from pre-defined template contracts. The `templateId` parameter refers to the template from which a data NFT or datatoken is created. The `templateId` can be set while creating data NFT/datatoken. Currently, Ocean protocol supports 1 template type for dataNFT and 2 template variants for datatokens, namely: **regular template** and **enterprise template**. Each template supports the same interface but differs in the underlying implementation and can have additional features.
+
+The details regarding currently supported datatoken templates are as follows:
+- **Regular template**: The regular template allows users to buy/sell/hold datatokens. The datatokens can be minted by the address having a `MINTER` role, making the supply of datatoken variable. 
```

Comment:
Can you add links to the smart contracts for each template

Reply:
We also need to explain the numbering, as this information is needed to mint the from the correct template. It's also how the templates are recorded in the sub graph.

Template 1 is the regular template and link to the [contract](https://github.com/oceanprotocol/contracts/blob/v4main/contracts/templates/ERC20Template.sol)

Template 2 is the enterprise template and also link to that [contract](https://github.com/oceanprotocol/contracts/blob/v4main/contracts/templates/ERC20TemplateEnterprise.sol).

Reply:
or we could just add description on how to call getId function (https://github.com/oceanprotocol/contracts/blob/v4main/contracts/templates/ERC20Template.sol#L844-L846  and  https://github.com/oceanprotocol/contracts/blob/v4main/contracts/templates/ERC20TemplateEnterprise.sol#L773-L775

Any app that is using the contracts should fetch that id and use the corresponding functions